packages/react/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -71,12 +71,14 @@ const CheckboxGroup = ({
 
   const [checkedNames, dispatch] = useReducer(reducer, checkedItems(items));
 
-  const itemsAsString = JSON.stringify(items);
-
-  useUpdate(
-    () => dispatch({ type: 'reset', state: checkedItems(items) }),
-    [itemsAsString],
-  );
+  const prevItems = usePrevious([...items]);
+  useUpdate(() => {
+    if (
+      items.length !== prevItems?.length ||
+      items.some((item, index) => !objectValuesEqual(item, prevItems[index]))
+    )
+      dispatch({ type: 'reset', state: checkedItems(items) });
+  });
```

Comment:
I assume its intended that the hook has no deps now?

Reply:
Yes. The comparison we want to do is now done in the `if` check above.